django-stubs/db/models/query.pyi

```diff
@@ -205,28 +208,28 @@ class RawQuerySet(Iterable[_T], Sized):
         hints: dict[str, Model] | None = ...,
     ) -> None: ...
     def __len__(self) -> int: ...
-    def __iter__(self) -> Iterator[_T]: ...
+    def __iter__(self) -> Iterator[_Model]: ...
     def __bool__(self) -> bool: ...
     @overload
-    def __getitem__(self, k: int) -> _T: ...
+    def __getitem__(self, k: int) -> _Model: ...
     @overload
     def __getitem__(self, k: str) -> Any: ...
     @overload
-    def __getitem__(self, k: slice) -> RawQuerySet[_T]: ...
+    def __getitem__(self, k: slice) -> RawQuerySet[_Model]: ...
     @cached_property
     def columns(self) -> list[str]: ...
     @property
     def db(self) -> str: ...
-    def iterator(self) -> Iterator[_T]: ...
+    def iterator(self) -> Iterator[_Model]: ...
     @cached_property
     def model_fields(self) -> dict[str, str]: ...
-    def prefetch_related(self, *lookups: Any) -> RawQuerySet[_T]: ...
+    def prefetch_related(self, *lookups: Any) -> RawQuerySet[_Model]: ...
     def resolve_model_init_order(self) -> tuple[list[str], list[int], list[tuple[str, int]]]: ...
-    def using(self, alias: str | None) -> RawQuerySet[_T]: ...
-
-_QuerySetAny: TypeAlias = _QuerySet  # noqa: PYI047
+    def using(self, alias: str | None) -> RawQuerySet[_Model]: ...
 
-QuerySet: TypeAlias = _QuerySet[_T, _T]
+# Obsolete aliases of QuerySet, for compatibility only.
```

Comment:
Do we really need this compat alias?

Reply:
Currently, yes. `ext/django_stubs_ext/aliases.py` still expects to import `from django.db.models.query import _QuerySet, _QuerySetAny`. I'll need to think if we change `aliases.py` without breaking compatibility.

It's really difficult to think about these, because the dependency constraint between django-stubs and django-stubs-ext only goes one way, users may be running e.g. `django-stubs-ext 5.0.1` and `django-stubs 5.0.0` together.

I wonder if we can turn the version constraint into `django-stubs-ext==5.0.0`, or will that cause issues for package managers?

Reply:
btw when adding review comments, please add it on the *last* relevant line of a change, rather than the first line, this way more of the diff context is visible.

Right now the message is missing the context that it's about `_QuerySet` and `_QuerySetAny`.

Reply:
Here's an idea:

*  #2117

We can get rid of `_QuerySetAny` after the above is merged. But the `_QuerySet` alias must remain for the time being.